superset/db_engine_specs/ocient.py

```diff
@@ -62,6 +64,11 @@
     "The reference to column '(?P<column>.*?)' is not valid"
 )
 
+# Store mapping of superset Query id -> cursor object
+# These are inserted into the cache when executing the query
+# They are then removed, either upon cancellation or query completion
+cursor_cache: Dict[Query, pyocient.Cursor]= dict()
```

Comment:
Still confused on why we need the Cursor object? Can we just store the UUID?

Reply:
At the time that the cursor object is placed into the cache, the UUID is None. We need to store the cursor object to be able to get the uuid after it is populated (upon sending the query to the database). We cannot store the id directly, as both Nonetypes and strings (our representation for the uuid in the backend) are immutable, and therefore are pass by value only. As a result, if we fetch the uuid now, it will not update when it is populated later.

Reply:
ah gotcha, that's a bummer!